cmd/main.go

```diff
@@ -359,8 +360,21 @@ func autoMigrationWithLeader(migrationLeader leader.ElectorInterface, db *gorm.D
 	return migrationLeader.RunWithLeader(context.Background(), func() error {
 		log.Infof("Start automigration")
 		err := db.AutoMigrate(&models.Host{}, &common.Cluster{}, &events.Event{}).Error
-		log.Infof("Finish automigration")
-		return err
+		if err != nil {
+			log.WithError(err).Fatal("Failed auto migration process")
```

Comment:
i think that fatal already terminate the application so it looks a bit strange to call fatal and return an error - but i don't know if there is  anything you can do about it. We definitely need to fail the application in this case.

Reply:
I could just remove the return here... but you're right that it's not immediately clear that `Fatal` would exit the app so maybe it's just worth having it for clarity?

Also I'm not sure why the `Fatal` was removed when this was moved into this leader function.

Reply:
It's a bad habit to have panic not inside main, because it synchronous operation the error should be handled in the main and panic there.